### PR TITLE
Updating InputCheckbox and InputSwitch to not need disableOnChange prop

### PR DESCRIPTION
**Pull request by jaruesink**

### CODE REVIEW

File: src/lib/inputs/InputCheckbox/InputCheckbox.tsx [hunk 1]
```diff
@@ -30,23 +28,12 @@ export const InputCheckbox = ({
   const [checked, setChecked] = useState(!!fieldProps?.value || !!props.checked);
```
Comment (jaredhill4) [inline src/lib/inputs/InputCheckbox/InputCheckbox.tsx:28]: So what we can do, I think is set a `fieldValue` variable at the top of the component scope:

```js
const fieldValue = !!fieldProps?.value || !!props.checked;
```

Then you can use that elsewhere, like here:

```js
const [checked, setChecked] = useState(fieldValue);
```

and here:

```js
useEffect(() => {
  setChecked(fieldValue);
}, [fieldValue]);
```

Then the question becomes, do you even need the `useEffect`? Or can you just use the `fieldValue`?

Comment (jaredhill4) [reply src/lib/inputs/InputCheckbox/InputCheckbox.tsx:28]: I just went through this with the `InputColor` component, and I was able to simplify it quite a bit, as well.

File: src/lib/inputs/InputSwitch/InputSwitch.tsx [hunk 2]
```diff
@@ -30,41 +29,24 @@ export const InputSwitch = ({
...
+    if (!checked && labelOff) setLabel(labelOff);
   }, [checked, labelOn, labelOff]);
 
   useEffect(() => {
-    if (disableOnChange) {
-      setChecked(!!fieldProps?.value || !!props.checked);
-    }
-  }, [disableOnChange, props.checked, fieldProps?.value]);
+    setChecked(!!fieldProps?.value || !!props.checked);
+  }, [props.checked, fieldProps?.value]);
```
Comment (jaredhill4) [inline src/lib/inputs/InputSwitch/InputSwitch.tsx:45]: Same here. See my comments from the `InputCheckbox` component.

File: src/lib/inputs/InputCheckbox/InputCheckbox.tsx (hunk 1 above)
Comment (jaruesink) [reply src/lib/inputs/InputCheckbox/InputCheckbox.tsx:28]: yeah let me try to use just fieldValue without an effect and see what happens!

**Pull request merged**